src/quickgui/plugin/qgsquickmapcanvas.qml

```diff
@@ -103,6 +103,31 @@ Item {
     mapCanvasWrapper.refresh()
   }
 
+  /**
+   * Animates movement of map canvas from oldPos to newPos.
+   * oldPos and newPos need to be in device pixels.
+   */
+  function jump( oldPos, newPos )
```

Comment:
@alexbruy , is there a specific need here to have oldPos instead of using the extent center point as oldPos? Using an oldPost that's not the current extent's center point would seemingly defeat the 'smooth' movement.

I'd also consider renaming the function to moveTo( pos ) or at least jump**To**( pos ). When I think of a jump to position, I'd also assume a slight zoom out then zoom in again (hence why I think moveTo would be a better function name).

Cool improvement :sunglasses:

Reply:
@alexbruy , opened this PR (https://github.com/qgis/QGIS/pull/50315) after realizing QGIS' QgsQuickMapSettings didn't yet have a center property. That should help you out here.